gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/AccountFactorsEndpointHandler.java

```diff
@@ -507,6 +507,51 @@ public void deleteRecoveryCode(RoutingContext routingContext) {
                 );
     }
 
+    /**
+     * Issue a challenge for the selected factor (SMS, Email mainly)
+     *
+     * @param routingContext the routingContext holding the current user
+     */
+    public void sendChallenge(RoutingContext routingContext) {
+        final User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        final String factorId = routingContext.request().getParam("factorId");
+
+        // find factor
+        findFactor(factorId, h -> {
+            if (h.failed()) {
+                routingContext.fail(h.cause());
+                return;
+            }
+
+            final FactorProvider factorProvider = factorManager.get(factorId);
+            if (factorProvider == null) {
+                routingContext.fail(new FactorNotFoundException(factorId));
+                return;
+            }
+
+            // get enrolled factor for the current user
+            Optional<EnrolledFactor> optionalEnrolledFactor = user.getFactors()
+                    .stream()
+                    .filter(enrolledFactor -> factorId.equals(enrolledFactor.getFactorId()))
+                    .findFirst();
+
+            if (!optionalEnrolledFactor.isPresent()) {
+                routingContext.fail(new FactorNotFoundException(factorId));
+                return;
+            }
+
+            final EnrolledFactor enrolledFactor = optionalEnrolledFactor.get();
+            sendChallenge(factorProvider, enrolledFactor, user, routingContext, sh -> {
```

Comment:
the sendChallenge method return a success if the factor doesn't manage the code emission.
As it is an explicit API call to send a code should we test `factorProvider.needChallengeSending()` before calling `sendChallenge` in order to throw a 400 status code (or 403 forbidden) in this specific case ?

Reply:
FP - 400 response is now the default behavior